### ISSUE TITLE
Cannot find 'OCKSurveyTaskViewController' in scope

### ISSUE THREAD

**Issue by zxcheergo** [carekit-apple/CareKit#645]
With the latest Xcode 13.3, I encounter the problem the Cannot find 'OCKSurveyTaskViewController' in scope. Actually, I import the  CareKit with the SPM. How to fix this problem? 
<img width="347" alt="image" src="https://user-images.githubusercontent.com/29516751/158421482-bb7e30e1-9039-44fa-9de9-296b5629e7ad.png">

**Comment by gavirawson-apple** [comment]
The `OCKSurveyTaskViewController` requires that ResearchKit is linked to your project. Can you confirm ResearchKit is linked? You can find linking instructions [here](https://github.com/ResearchKit/ResearchKit).

**Comment by zxcheergo** [comment]
Yes, I can confirm that I add the ResearchKit and import it where I use the OCKSurveyTaskViewController. It worked with the old version of Xcode 13.2.1. When I updated Xcode to version 13.3, first I got an error that "Module (ResearchKit) compiled with Swift 5.5.2 cannot be imported by the Swift 5.6 compiler".  Then I set Build Libraries for Distribution set to Yes for ResearchKit. After the operation, I build my project and get the error which I mentioned before.

**Comment by zxcheergo** [comment]
> The `OCKSurveyTaskViewController` requires that ResearchKit is linked to your project. Can you confirm ResearchKit is linked? You can find linking instructions [here](https://github.com/ResearchKit/ResearchKit).

Yes, I can confirm that I add the ResearchKit and import it where I use the OCKSurveyTaskViewController. It worked with the old version of Xcode 13.2.1. When I updated Xcode to version 13.3, first I got an error that "Module (ResearchKit) compiled with Swift 5.5.2 cannot be imported by the Swift 5.6 compiler". Then I set Build Libraries for Distribution set to Yes for ResearchKit. After the operation, I build my project and get the error that I mentioned before.

**Comment by zxcheergo** [comment]
> The `OCKSurveyTaskViewController` requires that ResearchKit is linked to your project. Can you confirm ResearchKit is linked? You can find linking instructions [here](https://github.com/ResearchKit/ResearchKit).

I also test the WWDC 21 RecoverApp and I also encounter this _can't find_ problem. It worked well with the old version of Xcode.

**Comment by cbaker6** [comment]
This is most likely related to #642. @zxcheergo you probably updated ResearchKit to latest commit on the main when updating Xcode.

If I'm right, you should be able to swap out your CareKit in SPM with the one on my branch, https://github.com/cbaker6/CareKit, specific commit `2e4bbe35162a0ecbe794afb543aca84067b6fd64` to get a working build. Basically, the CareKit minimum deployment needs to raise to iOS15 or the ResearchKit deployment needs to drop back to iOS13. You can also use the commit on ResearchKit right before the latest, ``

**Comment by zxcheergo** [comment]
> https://github.com/cbaker6/CareKit/tree/2.1-slider

Thanks for your replying! Could you please tell me how to use your branch 2.1-slider with the SPM in detail? Do I have to download and drag & drop in the project, instead of using SPM?

**Comment by cbaker6** [comment]
> Thanks for your replying! Could you please tell me how to use your branch 2.1-slider with the SPM in detail? Do I have to download and drag & drop in the project, instead of using SPM?

You use it just like you use the CareKit apple repo. Go to your Package Dependencies in your Xcode project, remove the current CareKit package, click the `+` button, paste https://github.com/cbaker6/CareKit/, for `Dependency Rule`, select `Branch`, and input 2.1-slider, press `Add Package`.

**Comment by zxcheergo** [comment]
> > Thanks for your replying! Could you please tell me how to use your branch 2.1-slider with the SPM in detail? Do I have to download and drag & drop in the project, instead of using SPM?
> 
> You use it just like you use the CareKit apple repo in SPM.
> 
> 1. Go to your `Package Dependencies` in your Xcode project
> 2. Remove the current CareKit package
> 3. Click the `+` button, paste `https://github.com/cbaker6/CareKit/` in the search field
> 4. For the `Dependency Rule`, select `Branch`, and input 2.1-slider
> 5. Press `Add Package`
> 6. Rebuild your app
> 
> You will need to use the same process when reverting back to the apple CareKit repo in the future

Thanks for your replying!

I follow your instruction and get the following error:

<img width="1079" alt="截屏2022-03-16 10 31 04" src="https://user-images.githubusercontent.com/29516751/158505003-ee153681-3d53-42a0-9eb6-03c582dd0da3.png">

**Comment by cbaker6** [comment]
You need to hit `enter` after you added the link in step 3.

You are currently searching for the 2.1-slider branch on Apple's repo in which it's not there

**Comment by zxcheergo** [comment]
> You need to hit `enter` pm your keyboard after you added the link in step 3.
> 
> You are currently searching for the 2.1-slider branch on Apple's repo in which it's not there. You can tell this is the case by looking at the link in your error above
> 
> The blue link to the CareKit repo should show my repo like below: ![image](https://user-images.githubusercontent.com/8621344/158505726-5371b1c2-05fe-45f9-8b5d-a49918126ce3.png)

It works!!! Thank you so much!

**Comment by samuelalake** [comment]
Please do you know why I get this message while trying to add package via SPM?

<img width="545" alt="Screen Shot 2022-03-21 at 10 44 56 PM" src="https://user-images.githubusercontent.com/58840187/159403822-115a727b-163a-4cb3-a6a0-23ee08220f01.png">

I'm unable to find any package result when I use the url

<img width="1077" alt="image" src="https://user-images.githubusercontent.com/58840187/159406358-9f68a77c-50a0-434d-bb7a-de87ea234e17.png">

**Comment by zxcheergo** [comment]
> You need to hit `enter` pm your keyboard after you added the link in step 3.
> 
> You are currently searching for the 2.1-slider branch on Apple's repo in which it's not there. You can tell this is the case by looking at the link in your error above
> 
> The blue link to the CareKit repo should show my repo like below: ![image](https://user-images.githubusercontent.com/8621344/158505726-5371b1c2-05fe-45f9-8b5d-a49918126ce3.png)

This problem happened again for no reason. I use the latest main branch of ResearchKit and the 2.1-slider of CareKit. Could you please help me! Thank you so much.

<img width="779" alt="image" src="https://user-images.githubusercontent.com/29516751/160443990-f204137c-1067-464c-8ca2-87f471a1a599.png">
<img width="204" alt="image" src="https://user-images.githubusercontent.com/29516751/160444031-a64517a1-1d23-4398-8073-77df4c5367d2.png">

**Comment by cbaker6** [comment]
> EDIT: It's most likely has to do with something it my Xcode's connection to Github. Couldn't fix.

Yes, this is something with your Xcode or your network connection. My repo is public, so anyone can read from it.

> This problem happened again for no reason. I use the latest main branch of ResearchKit and the 2.1-slider of CareKit. Could you please help me!

This most likely has something to do with how you are using Xcode (or if you updated to the latest Xcode version recently). Or, how you are importing `ResearchKit`. My CareKit repo won't cause these types of issues. You can try to [clean your build folder](https://developerinsider.co/clean-xcode-cache-quick-fix/).

**Comment by cbaker6** [comment]
If #642 gets merged developers can point directly to the Apple CareKit main branch to use the [latest commit of ResearchKit](https://github.com/ResearchKit/ResearchKit/pull/1499) with CareKit for iOS 13+.

Until then, you can:

1. Go to your `Package Dependencies` in your Xcode project
2. Remove the current CareKit package
3. Click the `+` button, paste `https://github.com/cbaker6/CareKit/` in the search field
4. For the `Dependency Rule`, select `Commit`, and input `ec157174bf77f95dcff53fd40d50bbfca5319b64`
5. Press `Add Package`
6. Rebuild your app
